lib/database/announcements.ts

```diff
@@ -0,0 +1,76 @@
+import { client } from "../db";
+import { Announcement } from "../../models";
+import { decode } from "io-ts-promise";
+import { array } from "io-ts";
+
+const AnnouncementArraySchema = array(Announcement);
+
+const getAnnouncements = async (classId: string): Promise<Announcement[]> => {
+  const query = {
+    text: "SELECT event_information_id, title, content, announcement_id FROM announcements WHERE event_information_id = $1",
+    values: [classId],
+  };
+  const res = await client.query(query);
+
+  try {
+    return await decode(AnnouncementArraySchema, res.rows);
+  } catch (e) {
+    throw Error("Fields returned incorrectly in database.");
+  }
```

Comment:
we dont need to try catch this anymore :). All the errors should just bubble to the top